src/unix.rs

```diff
@@ -174,41 +174,42 @@ impl Client {
                 //
                 // I tested this on macOS 14 and Linux 6.5.13
                 #[cfg(target_os = "linux")]
-                if let Ok(Some(jobserver)) =
+                if let Ok(Some(mut jobserver)) =
                     Self::from_fifo(&format!("fifo:/dev/fd/{}", read.as_raw_fd()))
                 {
+                    jobserver.path.take();
```

Comment:
This is fairly indirect in the way it's achieving its goal. At the very least it deserves a comment